src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2691,6 +2695,13 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
                 models=self.models.load_balancer_models,
             )
 
+        if self.context.get_nat_gateway_managed_outbound_ip_count() is not None:
+            network_profile.nat_gateway_profile = create_nat_gateway_profile(
+                self.context.get_nat_gateway_managed_outbound_ip_count(),
+                self.context.get_nat_gateway_idle_timeout(),
+                models=self.models.nat_gateway_models,
+            )
+
```

Comment:
Perhaps this part can be omitted as it's part of the create process, even if setting it to 0 is ignored, the default value will be backfilled to 0?

Reply:
The default value is 1